File: yourway-server/src/main/java/com/triador/yourwayserver/services/MarathonService.java
package com.triador.yourwayserver.services;

import com.triador.yourwayserver.dao.model.Marathon;

import java.util.List;

public interface MarathonService {

    Marathon saveMarathon(Marathon marathon);

    Marathon getMarathon(int marathonId);

    List<Marathon> getMarathons();
}

File: yourway-client/src/app/marathon/marathon.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';

import { Marathon } from '../models/marathon.model';

const marathonUrl = 'http://localhost:8080/marathons';

@Injectable()
export class MarathonService {

	constructor(private http: HttpClient) {}

	addMarathon(marathon: Marathon) {
		return this.http.post<Marathon>(marathonUrl, marathon);
	}

	getMarathons() {
		return this.http.get<Marathon[]>(marathonUrl);
	}

	getMarathon(marathonId: string) {
		return this.http.get<Marathon>(marathonUrl + '/' + marathonId);
	}
}